### PR TITLE
Allow explicit disable of in-app plugin

### PR DESCRIPTION
**Pull request by clabland**
`integrations: { "Customer.io In-App Plugin": { "enabled": false } }` will explicitly disable in-app.

---

> [!NOTE]
> **Low Risk**
> Low risk: small change to optional plugin registration logic plus new integration tests; main risk is unintentionally changing when the In-App plugin loads for edge-case settings objects.
> 
> **Overview**
> Updates `AnalyticsBrowser` plugin registration so the `Customer.io In-App Plugin` is only loaded when settings exist **and** `enabled !== false`, allowing callers to explicitly disable the plugin even if it’s present in CDN settings.
> 
> Extends `InAppPluginSettings` with an optional `enabled` flag and adds an integration test suite covering default CDN loading, explicit enable/disable via `options.integrations`, and overriding CDN-provided plugin settings.
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit 26303fb87f094bf391ebd1e4fcba7c8cbc05f510. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

### CODE REVIEW

**Review by BernardGatt — APPROVED** [review]

**Review by BernardGatt — COMMENTED** [review]
I gave this some thought and since we're planning to add this as part of the settings later on, shall we include it in the settings object instead of a plain `false`?

```
{
  "Customer.io In-App Plugin": {
    "enabled": false
  }
}
```

**Review by BernardGatt — CHANGES_REQUESTED** [review]

File: packages/browser/src/browser/index.ts [hunk 1]
```diff
@@ -210,22 +210,20 @@ async function registerPlugins(
...
+      | undefined)
+
+  const inAppPlugin =
+    inAppPluginSettings && inAppPluginSettings.enabled !== false
+      ? await import(
+          /* webpackChunkName: "inAppPlugin" */ '../plugins/in-app-plugin'
+        ).then((mod) => {
+          return mod.InAppPlugin(inAppPluginSettings)
+        })
+      : undefined
```
[the remark this answers is not in the document]
Comment (BernardGatt) [reply packages/browser/src/browser/index.ts:226]: Hmmm... I think cursor may be on to something here. Can you confirm @clabland ?

[the remark this answers is not in the document]
Comment (clabland) [reply packages/browser/src/browser/index.ts:226]: I think it's confused and was referencing my prior PR description (`integrations: { "Customer.io In-App Plugin": false }`), it works as expected in local testing.

**Review by BernardGatt — APPROVED** [review]
